### PR TITLE
Added plot recipe

### PR DESCRIPTION
**Pull request by hotplot**
Allows polynomials to be plotted using `plot(Poly([0, 1, -3, 2, -1]))` for example.

It adds `RecipesBase` as a dependency, but I am not sure how that should be handled since this repository does not have a `Project.toml`

### CODE REVIEW

File: test/REQUIRE [hunk 1]
```diff
@@ -1 +1,2 @@
-SpecialFunctions
\ No newline at end of file
+RecipesBase
```
Comment (martinholters) [inline test/REQUIRE]: Shouldn't this go to `REQUIRE` (as opposed to `test/REQUIRE`)? Or am I missing something?

Comment (hotplot) [reply test/REQUIRE]: I doubt you are missing anything; I am just not very familiar with pre-1.0 dependency management

**Comment by jverzani** [comment]
I tried plotting `x=variable(); p=(x-1)*(x-2)*(x-3)` and find the default range too wide and the command `plot(p, 0,4)` unavailable. I forget how the recipe works so that the range is computed only if a,b are not specified, but this seems like a necessary part of the interface.

**Comment by hotplot** [comment]
@jverzani I have added a second recipe which accepts two endpoints, rather than a range.

You now have the option of using an implicit range, explicit range, or explicit endpoints - `plot(p)`, `plot(p, 0 : 0.1 : 4)`, and `plot(p, 0, 4)` respectively.

I have also made the default range narrower.

**Comment by jverzani** [comment]
Thanks. Sorry about taking so long to merge.

**Pull request merged**